app/controllers/orders_controller.rb

```diff
@@ -11,10 +11,13 @@ class OrdersController < ApplicationController
 
   before_action :find_order, except: %i(index new create)
 
+  load_and_authorize_resource
```

Comment:
load_and_authorize_resource là ntn

Reply:
Cái này dùng để sử dụng được cancancan ạ, nó sẽ load resource cho class này, ví dụ e đặt ở orders_controller này thì cancancan sẽ xem người dùng hiện tại có được phép thực hiện hành động này (create read update destroy) cho model order không ạ

Reply:
chưa rõ em ơi, `load_and_authorize_resource` có chữ `and` kìa, tìm hiểu thêm xem nó bao gồm những logic xử lý nào nhé

Reply:
`load_and_authorize` dùng để load tất cả biến instance của các action CRUD và xác thực xem người dùng có đủ điều kiện thực hiện `:read, create, :update và :destroy `hay không đúng không anh

Reply:
- có thật là load tất cả các biến instance không ?
- đọc wiki thêm nhé em, tìm hiểu thêm `load_resource` và `authorize_resource`, action alias, .... trên wiki

Reply:
Em vừa đọc thì e hiểu nó dùng để thay thế cho cả 2 bước là `load_resource` và `authorize_resource`
`load_resource` dùng load các biến instance ví dụ như @order trong show thì sẽ = Order.find_by params[:id] và xác thực quyền CRUD của nó với người dùng hiện tại hay không, e hiểu được đến đó thôi anh à =)))

Reply:
Dùng load_resource sẽ tự tạo thêm biến instance cho hai hàm index vs show vs các quyền mình set trong can3 đấy. Nếu mà mình k cần thì bỏ load_resource cho đỡ tốn tài nguyên nhé

Reply:
Nghĩa là như nào ô nhi, trong trường hợp của tôi tôi vẫn muốn phân quyền cho nó cả 7 cái của resful này thì tôi dùng là đúng đúng k ?